src/commands/setmeme.ts

```diff
@@ -1,59 +1,43 @@
-import { Firestore } from '@google-cloud/firestore';
-import { Client, Message } from 'discord.js';
-
-exports.run = async (client: Client, msg: Message, args: string[], firestore: Firestore) => {
-  // Ensures only admins may use this command
-  if (!msg.member!.permissions.has('ADMINISTRATOR')) {
-    msg.channel.send('Error: insufficient permissions. Only Administrators may use this command.');
-    return;
-  }
-
-  let meme = false;
-
-  if (args === undefined || args.length === 0) {
-    const docRef = firestore.collection('guilds').doc(msg.guild!.id).collection('channels').doc(msg.channel.id);
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { CommandInteraction, GuildMember } from 'discord.js';
+import { Command } from '../types/command';
+import { CommandBody } from '../types/command-body';
+import { CommandOptions } from '../types/command-options';
+
+const data = new SlashCommandBuilder()
+	.setName('setmeme')
+	.setDescription('Sets the meme channel for stat tracking')
+  .addBooleanOption(option =>
```

Comment:
This command should either be reworked to take a channel reference, or just be baked into the 'setup' command.

Reply:
Will fix in https://github.com/Mr1upMachine/rusty-bot/issues/73